service-search-nls/src/main/java/fi/nls/oskari/search/util/ELFGeoLocatorCountries.java

```diff
@@ -31,53 +30,118 @@
     private long lastSync;
     private String serviceURL;
     private ELFGeoLocatorSearchChannel channel;
-    private Map<String, String> countries = new HashMap<>();
+    private static Map<String, String> countryMap = new HashMap<>();
 
-    public ELFGeoLocatorCountries(ELFGeoLocatorSearchChannel channel) {
-        this.channel = channel;
+    public ELFGeoLocatorCountries() {
         serviceURL = PropertyUtil.getOptional(ELFGeoLocatorSearchChannel.PROPERTY_SERVICE_URL);
         if (serviceURL == null) {
             log.warn("ServiceURL not configured. Add property with key", ELFGeoLocatorSearchChannel.PROPERTY_SERVICE_URL);
         }
+        loadCountryMap();
     }
 
-    public Map<String, String> getCountryMap() throws IOException {
-        // not properly configured
-        if (serviceURL == null) {
-            return countries;
+    public void loadCountryMap () {
+        try {
+            // not properly configured
+            if (serviceURL == null) {
+                return ;
+            }
+
+            // synced recently
+            if(System.currentTimeMillis() < lastSync + REFRESH_INTERVAL) {
+                return ;
+            }
+            parseCountryMap(getCountryData());
+            if(countryMap.isEmpty()) {
+                log.debug("Could not get countries");
+            }
         }
-        // synced recently
-        if(System.currentTimeMillis() < lastSync + REFRESH_INTERVAL) {
-            return countries;
+        catch (Exception e) {
+            log.debug(e, "Could not get countries");
         }
+        lastSync = System.currentTimeMillis();
```

Comment:
Don't update lastSync if the process fails

Reply:
Isn't this ok as it is, so if it fails it would possibly try to update all the time.

Reply:
L51 would force it to wait for the REFRESH_INTERVAL even if it fails